core/src/main/java/org/apache/iceberg/deletes/DeleteCounter.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.deletes;
+
+/** A counter to be used to count deletes as they are applied. */
+public class DeleteCounter {
+
+  private long count = 0L;
+
+  /** Increment the counter by one. */
+  public void increment() {
+    count++;
+  }
+
+  /** Return the current value of the counter. */
+  public long get() {
+    return count;
+  }
```

Comment:
I can’t recall if this was covered or not already, but is there a reason we can’t use `Counter` from the `metrics` package in api?

I think this work might have been done before the `Counter` class from api existed, but there’s also very likely a reason to stick with this simpler class (for Spark serde or something).

https://github.com/apache/iceberg/blob/master/api/src/main/java/org/apache/iceberg/metrics/Counter.java

Reply:
I’m not sure how the entire `MetricsContext` integrates with Spark and executors etc.

If we don’t have existing work that uses `MetricsContext` for custom Spark metrics (particularly task level metrics), possibly we can update this code in a later PR.

I know Steven handled a lot of the metrics context adjustments for Flink and @nastra did a lot of the work for that in general.

Reply:
This wasn't raised previously. I wasn't aware of `org.apache.iceberg.metrics.Counter` and in fact, it didn't exist when I started this work several months ago (although `org.apache.iceberg.metrics.MetricsContext.Counter` did exist).
I think `org.apache.iceberg.metrics.Counter` is a more complex API than needed here (in particular, we do not need a name or a unit), although one could use it. I'd prefer not to refactor the code. A more generic counter could be used, but `DeleteCounter` is specific to this use case and is for internal bookkeeping only.

Reply:
Agreed. And I can see how the MetricsContext work took place alongside this work (or realistically after it).

Unifying the two should be kept in mind, but the `DeleteCounter` is a well named, specific class that can’t really be mistaken for something else and can be updated later.

Thanks for clarifying @wypoon

Reply:
Yes, I think exposing this through `MetricsContext` could be valuable and such work would be a separate PR; at that time, the `DeleteCounter` could be replaced by the more general `Counter`.